app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java

```diff
@@ -168,65 +172,14 @@ private void rotateView(View v) {
+                "refreshCalendar:failureToRefresh"));
     }
 
-    private void pushEventAndDismiss(DialogInterface dialog, View dialogView, View v) {
-        Map<String, Object> data = new HashMap<>();
-        final String title = ((EditText) dialogView.findViewById(R.id.new_event_title)).getText().toString();
-        final String desc = ((EditText) dialogView.findViewById(R.id.new_event_desc)).getText().toString();
-        final String date = ((EditText) dialogView.findViewById(R.id.new_event_date)).getText().toString();
-        final String duration = ((EditText) dialogView.findViewById(R.id.new_event_duration)).getText().toString();
-        Map<String, String> event = new HashMap<>();
-        event.put("title", title);
-        event.put("desc", desc);
-        event.put("date", date);
-        event.put("duration", duration);
-        if (Util.putEventStringsInData(event, data)) {
-            dialog.dismiss();
-            return;
-        }
-        data.put("household", currentHouse);
-        db.collection("events").add(data)
-                .addOnSuccessListener(documentReference -> refreshCalendar(v));
-        dialog.dismiss();
-    }
-
-    @SuppressLint("NotifyDataSetChanged")
-    public void refreshCalendar(View v) {
-        db.collection("events")
-                .whereEqualTo("household", currentHouse)
-                .whereGreaterThan("start", LocalDateTime.now().toEpochSecond(ZoneOffset.UTC))
-                .get()
-                .addOnCompleteListener(task -> {
-                    if (task.isSuccessful()) {
-                        ArrayList<Event> newEvents = new ArrayList<>();
-                        for (QueryDocumentSnapshot document : task.getResult()) {
-                            // We assume the stored data is well behaved since it got added in a well behaved manner.
-                            Event event = new Event(
-                                    document.getString("title"),
-                                    document.getString("description"),
-                                    LocalDateTime.ofEpochSecond(document.getLong("start"), 0, ZoneOffset.UTC),
-                                    document.getLong("duration") == null ? 0 : document.getLong("duration"),
-                                    document.getId());
-                            newEvents.add(event);
-                        }
-                        calendarAdapter.notifyDataSetChanged();
-                        calendar.setEvents(newEvents);
-                    }
-                    else {
-                        Toast.makeText(v.getContext(), v.getContext().getString(R.string.refresh_calendar_fail), Toast.LENGTH_SHORT).show();
-                    }
-                });
-
+    private void addEvent(View v) {
+        calendar.addEvent(v, MainScreenActivity.this, db, currentHouse, calendarAdapter, Arrays.asList("MainScreenActivity",
+                "addEvent:failure"));
     }
```

Comment:
Could the other methods not be moved for some reason or do you think they belong here ? Im hesitant with them but feel like they might be able to be moved (initializeTaskList, registerForEventImage, loadHousehold, rotateView and rotateLists)

Reply:
I think the easier (or more difficult?) question is: what methods should be left in MainScreenActivity?

Reply:
My opinion is that initializeTaskList should be left in mainscreen, 
loadHousehold could be extracted but we would need to create a house class and since this method is only used here with a lot of internal variables I do not think it is beneficial.
I could move registerForEventImage to EventsAdapter but I'm not sure it would be beneficial, since you wrote the method here I'll leave it up to you to chose where and how it should be (in an other branch preferably).
I moved rotateView to the class Calendar.
I do not think rotateList should be moved since it manages both kinds of lists which are in different packages.

Reply:
This sounds good to me, ill eventually move registerForEventImage to EventsAdapter most likely (using the cleanup branch)